test/performance/odp_packet_gen.c

```diff
@@ -326,6 +376,85 @@ static int parse_vlan(const char *str, test_global_t *global)
 	return num_vlan;
 }
 
+static inline uint64_t bswap(uint64_t in, uint32_t len)
+{
+	uint8_t byte;
+	uint64_t result = 0;
+
+	if (ODP_BIG_ENDIAN || len == sizeof(uint8_t))
+		return in;
+
+	for (uint32_t i = 0; i < len; i++) {
+		byte = (in >> (8 * i)) & 0xff;
+		result |= ((uint64_t)byte << (8 * (len - 1 - i)));
+	}
+
+	return result;
+}
+
+static odp_bool_t parse_custom_fields(const char *optarg, test_options_t *opts)
+{
+	char *tmp_str = strdup(optarg), *tmp;
+	uint32_t num_fields = 0;
+	char name[MAX_HDR_NAME_LEN + 1];
+	uint32_t len;
+	uint64_t value;
+	int64_t diff;
+	hdr_field_t *hdr;
+	int ret;
+
+	if (tmp_str == NULL)
+		return false;
+
+	tmp = strtok(tmp_str, TOKEN_DELIMITER);
+
+	if (tmp == NULL) {
+		free(tmp_str);
+		return false;
+	}
+
+	opts->custom_l3.eth_type = strtol(tmp, NULL, 16);
+	tmp = strtok(NULL, TOKEN_DELIMITER);
+
+	while (tmp) {
+		if (num_fields == MAX_HDR_FIELDS) {
+			ODPH_ERR("Invalid custom header, too many fields: %u\n", num_fields + 1);
+			free(tmp_str);
+			return false;
+		}
+
+		ret = sscanf(tmp, "%" S(MAX_HDR_NAME_LEN) "[^" FIELD_DELIMITER "]" FIELD_DELIMITER
+			     "%u" FIELD_DELIMITER "%" PRIx64 FIELD_DELIMITER "%" PRIi64 "", name,
+			     &len, &value, &diff);
+
+		if (ret != 4) {
+			ODPH_ERR("Invalid custom header, bad field format\n");
+			free(tmp_str);
+			return false;
+		}
+
+		if (len > MAX_HDR_VALUE_SZ) {
+			ODPH_ERR("Invalid custom header, field length too long: %u\n", len);
+			free(tmp_str);
+			return false;
+		}
+
+		hdr = &opts->custom_l3.fields[num_fields];
+		odph_strcpy(hdr->name, name, MAX_HDR_NAME_LEN + 2);
```

Comment:
+2 due to truncation errors, added a comment about this in v4.

Reply:
In `odph_strcpy()` implementation won't the following line write the null char one byte too far considering `hdr_field_t.name` size is `MAX_HDR_NAME_LEN + 1`?
```
dst[sz - 1] = 0;
```

Reply:
@JereLeppanen Any ideas how to work around this issue? E.g. should ignore `-Wstringop-truncation` pragma be added to  `odph_strcpy()` implementation?

Reply:
I think adding the pragma to odph_strcpy() implementation would be fine. I was originally going to do that, but I didn't since it wasn't needed at the time. Same for _odp_strcpy().

I wonder though, why is it needed here and not anywhere else?

Reply:
In my quick testing `sscanf()` usage seemed to have some effect in this issue appearing.